cartographer/mapping_3d/sparse_pose_graph/optimization_problem.cc

```diff
@@ -154,6 +156,12 @@ void OptimizationProblem::Solve(const std::vector<Constraint>& constraints) {
             translation_parameterization(),
             common::make_unique<ceres::QuaternionParameterization>(), &problem);
       }
+      if (frozen) {
```

Comment:
Doesn't this mean that nodes could still move around as constraints are added to submaps in other trajectories?